### PR TITLE
fix pom file not found when run app with local engine

### LINKED ISSUE #48276: Local engine run on android_debug_unopt_x64 doesn't find POM
`flutter run --local-engine android_debug_unopt_x64` gives

`The file /Users/.../engine/src/out/android_debug_unopt_x64/x86_debug.pom wasn't found in the local engine out directory.`

But there's a x86_64_debug.pom file in android_debug_unopt_x64. Probably a tooling issue.

cc @blasten

### PR DESCRIPTION
**Pull request by zsunkun**
## Description
When I run app with local engine,I get the following error:
```
Launching lib/main.dart on MI 6X in debug mode...
The file /Users/xxx/engine/src/out/android_debug/arm64_v8a_debug.pom wasn't found in the local engine out directory.
```
 This is the content of my android_debug directory，it only has armeabi_v7a_debug.pom:
image

After fixed it, I got another problem:
```
* What went wrong:                                                      
Could not determine the dependencies of task ':app:compileDebugJavaWithJavac'.
> Could not resolve all task dependencies for configuration ':app:debugCompileClasspath'.
   > Could not find any matches for io.flutter:arm64_v8a_debug:+ as no versions of io.flutter:arm64_v8a_debug are available.
     Required by:                                                       
         project :app 
```
The reason for both issues is that getting the pom name and target-platform are both dependent on my phone(arm64).  I think when using a local Engine, it should depend on the engine out path

flutter dcotor:
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel v1.12.13-hotfixes, v1.12.13+hotfix.8, on Mac OS X 10.14.6 18G103, locale zh-Hans-CN)
[!] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    ✗ Android license status unknown.
      Try re-installing or updating your Android SDK Manager.
      See https://developer.android.com/studio/#downloads or visit https://flutter.dev/setup/#android-setup for detailed instructions.
[✓] Xcode - develop for iOS and macOS (Xcode 11.3)
[✓] Android Studio (version 3.5)
[✓] Connected device (1 available)
```
## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

Fixes https://github.com/flutter/flutter/issues/48276

### CODE REVIEW

**Comment by jonahwilliams** [comment]
FYI @xster @blasten

**Comment by blasten** [comment]
Thanks @zsunkun. This LG, but what about adding some unit tests in `gradle_test.dart`?

**Comment by jmagman** [comment]
`build_apk_test` test failures need to be addressed as well as new tests added.
https://cirrus-ci.com/task/4736705525710848?command=main#L818

**Comment by zsunkun** [comment]
> `build_apk_test` test failures need to be addressed as well as new tests added.
> https://cirrus-ci.com/task/4736705525710848?command=main#L818

@jmagman @blasten Unit tests have been updated

File: packages/flutter_tools/test/general.shard/android/gradle_test.dart [hunk 1]
```diff
@@ -1754,6 +1754,99 @@ plugin1=${plugin1.path}
       ProcessManager: () => mockProcessManager,
     });
 
+    testUsingContext(
+        'build apk uses selected local engine when connected devices and artifact have different abi', () async {
+      when(mockArtifacts.getArtifactPath(Artifact.flutterFramework,
+          platform: TargetPlatform.android_arm64, mode: anyNamed('mode'))).thenReturn('engine');
+      when(mockArtifacts.engineOutPath).thenReturn(fileSystem.path.join('out', 'android_arm'));
```
Comment (blasten) [inline packages/flutter_tools/test/general.shard/android/gradle_test.dart]: It might be worth testing each platform as well.

**Review by blasten — APPROVED** [review]
LGTM

**Pull request merged**